platform-validation/scripts/aziot-compatibility.sh

```diff
@@ -595,11 +657,34 @@ aziotedge_check() {
         SHARED_LIBRARIES="$(echo $SHARED_LIBRARIES "ld-linux-x86-64.so.2")"
     elif [ $ARCH = aarch64 ]; then
         SHARED_LIBRARIES="$(echo $SHARED_LIBRARIES "ld-linux-aarch64.so.1")"
-    elif [ $ARCH = armv7 ]; then
+    elif [ $ARCH = armv7l ]; then
         SHARED_LIBRARIES="$(echo $SHARED_LIBRARIES "ld-linux-armhf.so.3")"
     fi
+
     check_shared_library_dependency
 
+    eval binary_size='$'"$(echo "$ARCH"_iotedge_binaries_size)"
+    eval container_size='$'"$(echo "$ARCH"_iotedge_container_size)"
+    TOTAL_SIZE=$(echo $binary_size $container_size | awk '{print $1 + $2}')
+    if [ -z "$MOUNTPOINT" ]; then
+        wrap_debug "No Storage Path Provided, Using Present working directory"
+        MOUNTPOINT=$(pwd)
+    fi
+
+    check_storage_space "$MOUNTPOINT" "$TOTAL_SIZE" "$iot_edge_size_buffer"
+    ret="$?"
```

Comment:
why assign and use the exit code? why not directly check for it

Reply:
because the next executed statement clears $?, hence saving it